GHCJS cross build problems with servant

When trying to update a nix project to use the newest reflex-platform version found that I could not drop into nix-shell when using servant dependency. This only happened for reflex-platform versions above 0.4.0.0. Attempted minimal build and ran into same error. With only reflex as build-dep this dropped into shell as expected.

```nix
# default.nix
{ system ? builtins.currentSystem }:
(import ../reflex-platform { inherit system; }).project ({ pkgs, ... }: {
  packages = {
    common = ./common;
  };

  shells = {
    ghc = ["common"];
    ghcjs = ["common"];
  };
})

```

```cabal
-- common.cabal
name:                common
version:             0.1.0
build-type:          Simple
cabal-version:       >=1.10

library
  build-depends:  reflex
                , servant

executable common-exe
  main-is:             Main.hs
  build-depends:       common
```

```bash
sbe:~/Documents/reflex-test/test$ nix-shell -A shells.ghcjs
these derivations will be built:
  /nix/store/3q2rb2mrb5aspmd25wkawwpx3s8asksy-js-unknown-ghcjs-gdb-8.3.drv
  /nix/store/9244nq8x0bh31jzgkf36fnv3xclq94hs-python2.7-Cython-0.29.13.drv
  /nix/store/q5vf73k6d1pc7r9dgwznb1y3h6gdiw0x-python2.7-PyStemmer-1.3.0.drv
  /nix/store/rh12qzxnra32khv2f8xiznnvybj3l22i-python2.7-snowballstemmer-1.2.1.drv
  /nix/store/x73f3jk2hxy3kwrn5580glcm2dxcqd09-python2.7-sphinx-1.8.3.drv
  /nix/store/rw71jipm2ql7fpcsp6qqqacba7js3b4r-python2.7-readthedocs-sphinx-ext-0.5.17.drv
  /nix/store/q2vj4zibm1z1fshw13yynqsy3d65ib7l-python2.7-sphinx_rtd_theme-0.4.3.drv
  /nix/store/xh40grz9mpa65mn78dk7j5sbqnv2wgd9-python2.7-recommonmark-0.5.0.drv
  /nix/store/l76fnkyq1x9yj84fsk6idz2dkljliysr-servant-sphinx-documentation-0.16.2-js-unknown-ghcjs.drv
  /nix/store/wdwamdwvg7ldpd9gqvn7va5zx7jcha5k-servant-0.16.2-js-unknown-ghcjs.drv
  /nix/store/vc3phbx2jky04gda549h8asiksim1a0x-hoogle-local-0.1.drv
  /nix/store/n36cr56ijjm5c1gv5yyj7drfac41hchn-ghcjs-8.6.0.1-with-packages.drv
building '/nix/store/3q2rb2mrb5aspmd25wkawwpx3s8asksy-js-unknown-ghcjs-gdb-8.3.drv'...
unpacking sources
unpacking source archive /nix/store/5ipja010hc63kvh16vbs4qdn8h4gakz4-gdb-8.3.tar.xz
source root is gdb-8.3
setting SOURCE_DATE_EPOCH to timestamp 1557598931 of file gdb-8.3/md5.sum
patching sources
applying patch /nix/store/k8x7m6j9lqzdmasqvsrdhh5n009pnwnl-debug-info-from-env.patch
patching file gdb/main.c
Hunk #1 succeeded at 556 (offset 5 lines).
configuring
fixing libtool script ./ltmain.sh
configure flags: --prefix=/nix/store/p9cpryzfcaijgk3iaym3jl4kdl3xkgp5-js-unknown-ghcjs-gdb-8.3 --enable-targets=all --enable-64-bit-bfd --disable-install-libbfd --disable-shared --enable-static --with-system-zlib --with-system-readline --with-gmp=/nix/store/p9swjnshwdd3h5n0s71q8n0nyybfq3km-gmp-6.1.2-dev --with-mpfr=/nix/store/8aw5qh2ynvn5wv9k51a7d0ip9alf8ksr-mpfr-4.0.2-dev --with-expat --with-libexpat-prefix=/nix/store/bqp30vkncmm222mjvwggz0s7p318sflj-expat-2.2.7-dev --build=x86_64-unknown-linux-gnu --host=x86_64-unknown-linux-gnu --target=js-unknown-ghcjs
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... Invalid configuration `js-unknown-ghcjs': machine `js-unknown' not recognized
configure: error: /nix/store/506nnycf7nk22x7n07mjjjl2g8nifpda-bash-4.4-p23/bin/bash ./config.sub js-unknown-ghcjs failed
builder for '/nix/store/3q2rb2mrb5aspmd25wkawwpx3s8asksy-js-unknown-ghcjs-gdb-8.3.drv' failed with exit code 1
cannot build derivation '/nix/store/9244nq8x0bh31jzgkf36fnv3xclq94hs-python2.7-Cython-0.29.13.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/q5vf73k6d1pc7r9dgwznb1y3h6gdiw0x-python2.7-PyStemmer-1.3.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/rh12qzxnra32khv2f8xiznnvybj3l22i-python2.7-snowballstemmer-1.2.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/x73f3jk2hxy3kwrn5580glcm2dxcqd09-python2.7-sphinx-1.8.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/l76fnkyq1x9yj84fsk6idz2dkljliysr-servant-sphinx-documentation-0.16.2-js-unknown-ghcjs.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/wdwamdwvg7ldpd9gqvn7va5zx7jcha5k-servant-0.16.2-js-unknown-ghcjs.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/n36cr56ijjm5c1gv5yyj7drfac41hchn-ghcjs-8.6.0.1-with-packages.drv': 1 dependencies couldn't be built
error: build of '/nix/store/n36cr56ijjm5c1gv5yyj7drfac41hchn-ghcjs-8.6.0.1-with-packages.drv' failed

```

--- a/haskell-overlays/ghcjs.nix
+++ b/haskell-overlays/ghcjs.nix
@@ -61,6 +61,12 @@ self: super: {
   QuickCheck = dontCheck super.QuickCheck;
   temporary = dontCheck super.temporary;
 
+  # Custom docs don't build with ghcjs
+  # https://github.com/reflex-frp/reflex-platform/issues/631
+  servant = haskellLib.overrideCabal super.servant (drv: {
+    postInstall = "";
+  });
+
   # These tests never complete
   tasty-quickcheck = dontCheck super.tasty-quickcheck;
 

